### PR TITLE
Issue/164 add preformat span

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Add support for preformat span using the `<pre>` tag as described in https://github.com/wordpress-mobile/AztecEditor-Android/issues/164.

#### Note
Consecutive whitespace characters are removed from block spans.  See below for example.  That will be addressed in https://github.com/wordpress-mobile/AztecEditor-Android/issues/322.

***HTML***
```
<pre>
when (person) {<br>
    MOCTEZUMA -> {<br>
        print (\"friend\")<br>
    }<br>
    CORTES -> {<br>
        print (\"foe\")<br>
    }<br>
}
</pre>
```

***Visual***
```
when (person) {
MOCTEZUMA -> {
print ("friend")
}
CORTES -> {
print ("foe")
}
}
```

### Test
Enter text within a `<pre></pre>` tag to apply preformat formatting in HTML mode.  Select ***Preformat*** option from the heading menu in the format toolbar to apply preformat formatting in visual more.

### CODE REVIEW

File: aztec/src/main/kotlin/org/wordpress/aztec/formatting/BlockFormatter.kt [hunk 1]
```diff
@@ -646,12 +666,62 @@ class BlockFormatter(editor: AztecText, val listStyle: ListStyle, val quoteStyle
...
-                TextFormat.FORMAT_HEADING_6)
-                .filter { it != textFormat }
+                TextFormat.FORMAT_HEADING_6,
+                TextFormat.FORMAT_PREFORMAT)
+            .filter { it != textFormat }
 
         return containsHeading(textFormat, selStart, selEnd) && otherHeadings.none { containsHeading(it, selStart, selEnd) }
     }
 
+    fun containPreformat(selStart: Int = selectionStart, selEnd: Int = selectionEnd): Boolean {
```
Comment (0nko) [inline aztec/src/main/kotlin/org/wordpress/aztec/formatting/BlockFormatter.kt]: nitpick: other methods start with "contains" instead of "contain"

**Comment by theck13** [comment]
> I found some weird formatting behavior

That should be fixed with c9ef7f53015cde79fd9962779720cbca1b323482.

**Comment by 0nko** [comment]
I played with it a bit and found a glitch:

1. Tap on "CORTES" inside the `<pre>` section
2. Apply H3 from the heading menu
3. Apply Paragraph from heading menu
4. The part above "CORTES" has H3 style = OK
5. The "CORTES" party has no heading applied = OK
6. Notice the part below "CORTES" has H1 style applied

**Comment by theck13** [comment]
That looks like a problem with heading elements.  If you try the steps below on `develop`, you get the same result.  I'll create a new issue for it.

### Steps
1. Enter the following in HTML mode.
```
<h3>First
Second
Third
</h3>
```
2. Place cursor anywhere in "Second" text.
3. Tap ***Heading*** format button.
4. Tap ***Paragraph*** menu option.
5. Notice "First" is Heading 3 as expected.
6. Notice "Second" is Paragraph as expected.
7. Notice "Third" is Heading 1 which is unexpected.

**Comment by 0nko** [comment]
:shipit:

**Pull request merged**